### PR TITLE
fix: Support memoization on plugin node. Fixes #8553

### LINKED ISSUE #8553: Memoization not work on plugin node.
## Summary
Memoization cache is not generated on plugin node.
## Diagnostics
Paste the smallest workflow that reproduces the bug. We must be able to run the workflow.
Try to give an example below. We use an inner plugin, here gives a mock plugin.
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  name: hello-plugin-demo
spec:
  entrypoint: main
  templates:
    - name: main
      dag:
        tasks:
          - name: hello1
            template: hello1
    - name: hello1
      plugin:
        hello: { }
      memoize:
        key: "hello-plugin-demo-hello1"
        maxAge: "3600s"
        cache:
          configMap:
            name: hello-plugin-demo-hello1
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by wujayway**
Signed-off-by: wujayway <wujayway@gmail.com>

Fixes #8553

### CODE REVIEW

File: workflow/controller/taskset.go [hunk 1]
```diff
@@ -134,6 +137,13 @@ func (woc *wfOperationCtx) reconcileTaskSet(ctx context.Context) error {
 			node.FinishedAt = metav1.Now()
 
 			woc.wf.Status.Nodes[nodeID] = node
+			if node.MemoizationStatus != nil && node.Succeeded() {
```
Comment (alexec) [inline workflow/controller/taskset.go:140]: test?

Comment (wujayway) [reply workflow/controller/taskset.go:140]: > test?

done

**Comment by wujayway** [comment]
Hi @alexec ,
ut not work after merge #8557.
I can pass ut on local.
Not sure what happened, need help.

File: workflow/controller/taskset_test.go [hunk 2]
```diff
@@ -333,3 +334,175 @@ func TestNonHTTPTemplateScenario(t *testing.T) {
 		assert.NoError(t, err)
 	})
 }
+
+func TestReconcileTaskSetWithMemoization(t *testing.T) {
+	wf := wfv1.MustUnmarshalWorkflow(`apiVersion: argoproj.io/v1alpha1
+kind: Workflow
```
Comment (alexec) [inline workflow/controller/taskset_test.go:340]: This example makes it too hard to understand what is the example is relevant to the test, and what is just copied and pasted. Can you please simplify.

Comment (wujayway) [reply workflow/controller/taskset_test.go:340]: ok

File: pkg/apis/workflow/v1alpha1/workflow_types_test.go [hunk 3]
```diff
@@ -829,24 +829,40 @@ func TestWorkflow_SearchArtifacts(t *testing.T) {
 	queriedArtifactSearchResults := wf.SearchArtifacts(query)
 	assert.NotNil(t, queriedArtifactSearchResults)
 	assert.Len(t, queriedArtifactSearchResults, 3)
-	assert.Equal(t, "artifact-foo", queriedArtifactSearchResults[0].Artifact.Name)
```
Comment (alexec) [inline pkg/apis/workflow/v1alpha1/workflow_types_test.go:832]: ?

Comment (alexec) [reply pkg/apis/workflow/v1alpha1/workflow_types_test.go:832]: what is this please?

Comment (alexec) [reply pkg/apis/workflow/v1alpha1/workflow_types_test.go:832]: @rohankmr414 maybe this is the fix?

Comment (rohankmr414) [reply pkg/apis/workflow/v1alpha1/workflow_types_test.go:832]: yeah looks like it

**Review by alexec — APPROVED** [review]

**Pull request merged**